ports/stm32/powerctrl.c

```diff
@@ -713,6 +714,13 @@ void powerctrl_enter_stop_mode(void) {
     }
     #endif
 
+    #if MICROPY_HW_ENABLE_USB
+    bool usb_connected = pyb_usb_dev_connected();
+    if (usb_connected) {
+        pyb_usb_dev_disconnect();
+    }
+    #endif
```

Comment:
I think this should be called at the very start of this function, before IRQs are disabled.  It's a high level operation.